kernel/core/datamanagement/memory.cpp

```diff
@@ -3748,94 +3748,162 @@ std::vector<mu::value_type> Memory::getIndex(size_t col, const std::vector<mu::v
+std::vector<AnovaResult> Memory::getAnova(const VectorIndex& colCategories, size_t colValues, const VectorIndex& _vIndex, double significance) const
 {
     // Get indices
-    if (colCategories > memArray.size()
-        || !memArray[colCategories]
-        || memArray[colCategories]->m_type != TableColumn::TYPE_CATEGORICAL
+    if (colCategories[0] > memArray.size()
+        || !memArray[colCategories[0]]
+        || memArray[colCategories[0]]->m_type != TableColumn::TYPE_CATEGORICAL
         || significance >= 1.0
-        || significance <= 0.0)
+        || significance <= 0.0
+        || colCategories.size() > 2)
     {
         AnovaResult res;
         res.m_FRatio = NAN;
-        return res;
+        return std::vector<AnovaResult> {res};
     }
 
-    _vIndex.setOpenEndIndex(getElemsInColumn(colCategories)-1);
-    Memory _mem(2);
-    _mem.memArray[0].reset(memArray[colCategories]->copy(_vIndex));
-    _mem.memArray[1].reset(memArray[colValues]->copy(_vIndex));
+    bool isTwoWay = colCategories.size() == 2;
+    _vIndex.setOpenEndIndex(getElemsInColumn(colCategories[0])-1);
+
+    Memory _mem(colCategories.size()+1);
+    _mem.memArray[0].reset(memArray[colValues]->copy(_vIndex));
+    for(int i = 0; i < colCategories.size(); i++)
+        _mem.memArray[i+1].reset(memArray[colCategories[i]]->copy(_vIndex));
+
+    std::vector<std::vector<std::string>> factors;
+    for(int i = 0; i < colCategories.size(); i++)
+        factors.push_back(static_cast<CategoricalColumn*>(_mem.memArray[i+1].get())->getCategories());
 
-    const std::vector<std::string>& vCategories = static_cast<CategoricalColumn*>(_mem.memArray[0].get())->getCategories();
+    std::vector<Memory> groupedValues(std::pow(2, factors.size()) - 1);
 
-    // Copy into different columns
-    for (const auto& cat : vCategories)
+    // Copy the gropues into different columns (g1, g2, g1Xg2)
+    // For 3+ - way anova here some algo for clustering in all subsets of combinations of groups would be needed
+    for (int i = 0; i < factors[0].size(); i++)
     {
-        std::vector<mu::value_type> catIndex = _mem.getIndex(0, std::vector<mu::value_type>(), std::vector<std::string>(1, cat));
+        //positions of all elements, which correspond to the passed values
+        std::vector<mu::value_type> catIndex1 = _mem.getIndex(1, std::vector<mu::value_type>(), std::vector<std::string>(1, factors[0][i]));
 
-        if (mu::isnan(catIndex.front()))
+        if (mu::isnan(catIndex1.front()))
             continue;
 
-        _mem.memArray.push_back(TblColPtr(_mem.memArray[1]->copy(VectorIndex(&catIndex[0], catIndex.size(), 0))));
+        groupedValues[0].memArray.push_back(TblColPtr(_mem.memArray[0]->copy(VectorIndex(&catIndex1[0], catIndex1.size(), 0))));
+
+        // will not be triggered if one-way anova
+        for (int j = 0; isTwoWay & j < factors[1].size(); j++)

```

Comment:
```suggestion
        for (size_t j = 0; isTwoWay && j < factors[1].size(); j++)
```
That was very likely not intended to be a "binary and"